plugins/amazonq/shared/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/lsp/AmazonQLspService.kt

```diff
@@ -265,7 +275,23 @@ private class AmazonQServerInstance(private val project: Project, private val cs
         launcherHandler.addProcessListener(inputWrapper)
         launcherHandler.startNotify()
 
-        launcher = LSPLauncher.Builder<AmazonQLanguageServer>()
+        class AmazonQServerBuilder : Builder<AmazonQLanguageServer>() {
+            private val customMessageTracer = MessageTracer()
```

Comment:
how is this different than the existing logger?

Reply:
We need a custom one since the overriden message consumer does not respect the preexisting one

Reply:
```
launcher = LSPLauncher.Builder<AmazonQLanguageServer>()
        .wrapMessages { consumer ->
            MessageConsumer { message ->
                // custom logic here

                // finally consume
                consumer.consume(message)
            }
        }
```